### ISSUE TITLE
Allow set a GoogleTranslate host

### ISSUE THREAD

**Issue by linsui** [dialect-app/dialect#259]
In China mainland, only translate.google.cn is accessible. I'd like to only use it instead of a random host. Thanks!

**Comment by fushinari** [comment]
Would you consider using your location to apply necessary workarounds to be a decent solution for this?

**Comment by linsui** [comment]
Do you mean selecting the host based on the system location? I would prefer that it's an option.

**Comment by rafaelmardojai** [comment]
@linsui The problem is that the random host is part of the approach to avoid posible Google bans. So exposing this setting to everyone might be a door to problems.

One "invisible" approach that we were thinking was to ask for geolocation access (using the `org.freedesktop.portal.Location` portal) in the first run so we can apply conditional fixes for the user country.

**Comment by linsui** [comment]
Got it! That makes sense. Would it be easier to just check the conectivity? My concern is that some Chinese users may have proxy setup and don't want to use google.cn.

**Comment by rafaelmardojai** [comment]
> Would it be easier to just check the conectivity?

Right, I guess we could look for the better approach to detect this without user iteration when we get to it.

**Comment by rafaelmardojai** [comment]
> My concern is that some Chinese users may have proxy setup and don't want to use google.cn

Also probably we will have a prompt before asking for the location, so they could just dismiss it.

**Comment by linsui** [comment]
[Google shutdown the service in China.](https://techcrunch.com/2022/09/30/google-appears-to-have-disabled-google-translate-in-parts-of-china/) So this is not useful to me anymore. :shrug:

**Comment by fushinari** [comment]
> [Google shutdown the service in China](https://techcrunch.com/2022/09/30/google-appears-to-have-disabled-google-translate-in-parts-of-china/)

Oh... sorry to hear that
